step/matcher/staged_pipeline.go

```diff
@@ -9,39 +9,39 @@ import (
 
 const DELIMITER = "."
 
-type BuildIDGetter struct {
-	FinishedStages model.FinishedStages
-	TargetNames    []string
-	logger        log.Logger
+type stagedPipelineMatcher struct {
+	finishedStages model.FinishedStages
+	targetNames    []string
+	logger         log.Logger
 }
 
 type keyValuePair struct {
 	key   string
 	value string
 }
 
-func NewBuildIDGetter(finishedStages model.FinishedStages, targetNames []string, logger log.Logger) BuildIDGetter {
-	return BuildIDGetter{
-		FinishedStages: finishedStages,
-		TargetNames:    targetNames,
-		logger: logger,
+func newStagedPipelineMatcher(finishedStages model.FinishedStages, targetNames []string, logger log.Logger) stagedPipelineMatcher {
+	return stagedPipelineMatcher{
+		finishedStages: finishedStages,
+		targetNames:    targetNames,
+		logger:         logger,
 	}
 }
 
-func (bg BuildIDGetter) GetBuildIDs() ([]string, error) {
+func (s stagedPipelineMatcher) Matches() ([]string, error) {
```

Comment:
you missed the opportunity to call this `spm`, ("Super Pursuit Mode", a feature of KITT from Knight Rider)

Reply:
Hahaha, I like this and made it an easter egg for the future step developers. 😄